parallel/runner.go

```diff
@@ -145,3 +150,34 @@ func (r *runner) Cancel() {
 func (r *runner) Errors() map[int]error {
 	return r.errors
 }
+
+// Define the work as done when all consumers are idle for idleThresholdSeconds.
+// Can be run by the producer as a go routine right after starting to produce.
+// CAUTION - Might panic if tasks are taking more than idleThresholdSeconds to be produced.
+func (r *runner) DoneWhenAllIdle(idleThresholdSeconds int) error {
+	for {
+		time.Sleep(time.Duration(idleThresholdSeconds) * time.Second)
```

Comment:
Probably better to move the above sleep line to be at the end of the loop, rathen than at beginning, to allow for a possible situation where no sleep is needed.

Reply:
Needed to avoid an instant panic when the producer consumer is initialized.